src/yb/docdb/docdb_rocksdb_util.cc

```diff
@@ -487,8 +507,16 @@ void InitRocksDBOptions(
     options->num_reserved_small_compaction_threads = FLAGS_num_reserved_small_compaction_threads;
   }
 
-  options->compression = rocksdb::Snappy_Supported() && FLAGS_enable_ondisk_compression
-      ? rocksdb::kSnappyCompression : rocksdb::kNoCompression;
+  rocksdb::CompressionType compressionType = rocksdb::kNoCompression;
+  if (FLAGS_enable_ondisk_compression){
+      if (FLAGS_compression_type == 0 && rocksdb::LZ4_Supported()){
+          compressionType = rocksdb::kLZ4Compression;
+      }else if (FLAGS_compression_type == 1 && rocksdb::Snappy_Supported()){
+          compressionType = rocksdb::kSnappyCompression;
+      }
```

Comment:
if we configure a compression type which is not supported, we should probably **_at least_** `LOG(ERROR)`